guides/image-sources/provisioning-image-cloudinit.plantuml

```diff
@@ -0,0 +1,55 @@
+@startuml
+
+!include foreman.pstyle
+
+title Image-based provisioning with cloud-init configuration
+
+actor       User
+participant "Provisioned\nInstance" as Host
+participant "Foreman" as Foreman
+participant "Foreman\nProxy" as Proxy
+participant "Infrastructure\nCloud" as Cloud
+participant DNS
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Foreman : has an image\nwith cloud-init flag
+
+== Create host in Foreman ==
+
+User -> Foreman : clicks **Create Host**
+User -> Foreman : selects cloud resource,\nthe image and submits
+group Template [cloud-init]
+   Foreman -> Proxy : renders the cloud-init script
+end
+Foreman -> Cloud : creates new instance\nwith cloud-init info
+Cloud -> Foreman : reports IP address
+Foreman -> Proxy : creates DNS records
+Proxy -> DNS : forwards DNS records
+!if ($puppet)
+Foreman -> Proxy : creates Puppet sign request (autosign enabled)
+Proxy -> Puppet : forwards Puppet sign request
+!endif
+Foreman -> Cloud : starts the instance
+note over Host : boots up
+
+== Cloud-init script ==
+
+!include prov-initial-configuration.iuml
+Host -> Foreman : calls home\n(disables build mode)
+
+!if ($puppet)
+== First Puppet run ==
+
+Host -> Puppet : sends initial facts
+Puppet -> Foreman : forwards initial facts
+Host -> Puppet : compiles catalog
+note over Host : host is configured
```

Comment:
I'd say this is different, depending on how much detail you want to include.

If the host bootstrapped, we can assume a certificate. Our default templates do not include this, so it first starts with the SSL bootstrap:
* Host -> Puppet : send CSR
* Puppet -> Host : send certificate (if allowed to)

Then a regular Puppet run continues:

* Host -> Puppet : send facts
* Host -> Puppet : request catalog
* Puppet -> Foreman : forward facts
* Puppet -> Foreman : request ENC
* Puppet -> Host : respond with requested catalog
* Host runs catalog
* Host -> Puppet : send report
* Puppet -> Foreman : forward report

Reply:
Isn't the SSL bootstrap part of initial configuration, i.e. `prov-initial-configuration.iuml`?